.github/workflows/interop-test.yml

```diff
@@ -2,15 +2,22 @@ on: [push, pull_request]
 name: Interoperability Testing
 
 jobs:
+  # NOTE: during a pull request run, github creates a merge commit referenced in `github.sha`
+  # that merge commit is not a regular commit. You won't find it with a regular `git checkout SHA` and
+  # tools like `go get repo@SHA` won't find it.
+  #
+  # As a workaround, we generate a path to the actual pull request's commit, it looks like:
+  # `github.com/external-org/go-libp2p@latest-commit-on-their-branch`
   run-ping-interop-cross-version:
     uses: "libp2p/test-plans/.github/workflows/run-composition.yml@master"
     with:
       composition_file: "ping/_compositions/go-cross-versions.toml"
-      # NOTE: during a pull request run, github creates a merge commit referenced in `github.sha`
-      # that merge commit is not a regular commit. You won't find it with a regular `git checkout SHA` and
-      # tools like `go get repo@SHA` won't find it.
-      #
-      # As a workaround, we generate a path to the actual pull request's commit, it looks like:
-      # `github.com/external-org/go-libp2p@latest-commit-on-their-branch`
       custom_git_target: github.com/${{ github.event.pull_request.head.repo.full_name || github.event.repository.full_name }}
       custom_git_reference: ${{ github.event.pull_request.head.sha || github.sha }}
+  run-ping-interop-cross-implementation:
```

Comment:
Is this the test that takes 70 minutes that you talked about yesterday?

Reply:
No no, there is no test that takes 70 minutes,

This test takes ~17 minutes. It test go-libp2p master + rust-libp2p master + the current branch.
There is a larger test that takes around ~27 minutes that tests all known versions together + the current branch but we won't enable it yet.

See the results here:
https://github.com/libp2p/test-plans/pull/32#issuecomment-1225994414

Reply:
I shared 2 examples pull requests in the description if you want to see a recent run using this code:

passing: https://github.com/laurentsenta/go-libp2p/actions/runs/2955179619
breaking: https://github.com/laurentsenta/go-libp2p/actions/runs/2955183270